src/modbus/ModbusClient.cpp

```diff
@@ -40,12 +44,18 @@ bool ModbusClient::connect()
         return false;
     }
 
-    if (modbus_connect(m_modbus) == -1)
+    LOG(ERROR) << "LibModbusClient: Attempting to connect";
+    while (modbus_connect(m_modbus) == -1)
     {
         LOG(ERROR) << "LibModbusClient: Unable to connect - " << modbus_strerror(errno);
-        destroyContext();
-        return false;
+        std::this_thread::sleep_for(std::chrono::seconds(m_timeoutDurations[m_timeoutIterator]));
```

Comment:
It was agreed that reconnect logic is to be moved to the caller of connect (ModbusBridge), as no method call should block especially not for few hours